File: Basic _c_program/.vscode/Python codes/DS lab codes/naivepattern.c
#include <stdio.h>
#include <string.h>

void native_pattern(const char* txt, const char* pat) {
    int m = strlen(txt);
    int n = strlen(pat);
   
    for (int i = 0; i <= m - n; i++) {
        int j;
        for (j = 0; j < n; j++) {
            if (txt[i + j] != pat[j]) {
                break;
            }
        }
        if (j == n) {
            printf("Pattern found at index %d\n", i);
        }
    }
}

int main() {
    char txt[100];
    char pat[100];

    printf("Enter the text: ");
    scanf("%s", txt);

    printf("Enter the pattern: ");
    scanf("%s", pat);

    native_pattern(txt, pat);

    return 0;
}

File: Basic _c_program/.vscode/Python codes/DS lab codes/naivepattern.py
def Naive_pattern(txt,pat):
    m=len(txt)
    n=len(pat)
    for i in range(0,m-n):
        for j in range(0,n):
            if (txt[i+j]!=pat[j]):
                break
            if j==n-1:
                print("PAtter found at",i)
                
txt='HSYSDVBFUF'
pat='SDV'
Naive_pattern(txt,pat)